### PR TITLE
Add test for boilerplate to package_sanity

### PR DESCRIPTION
**Pull request by baberlevi**
This addresses #7109

Depends on #9282 #9283 #9284, these are all packages that would fail the test currently.

### CODE REVIEW

**Comment by baberlevi** [comment]
re-triggering travis

**Comment by baberlevi** [comment]
Travis not timing out anymore, but this new test does fail as expected (the existing packages which contain boilerplate need merged first)

**Comment by baberlevi** [comment]
triggering travis again now that deps are merged

File: lib/spack/spack/test/package_sanity.py [hunk 1]
```diff
@@ -78,3 +78,20 @@ def test_package_version_consistency():
...
+def test_no_fixme():
+    """Packages should not contain any boilerplate such as
+       FIXME or example.com."""
+    errors = []
+    for name in spack.repo.all_package_names():
+        repo = spack.repo.Repo(spack.paths.packages_path)
+        filename = repo.filename_for_package_name(name)
+        with open(filename, 'r') as package_file:
+            for line in package_file:
+                if (re.search(r'remove this boilerplate', line) or
```
Comment (tgamblin) [inline lib/spack/spack/test/package_sanity.py]: Can the list of regexes be factored out?  And can we have error message that tell the user what to fix?

```python

fixme_regexes = [
    r'remove this boilerplate',
    r'FIXME: Put',
    r'FIXME: Add',
    r'example.com',
]

errors = []
for i, line in enumerate(package_file):
    pattern = next((r for r in fixme_regexes if re.search(r, line)), None)
    if pattern:
        errors.append("%s:%d: boilerplate needs to be removed: %s" % (filename, i, line.strip()))
assert [] == errors    # pytest should print a nice diff for this.
```

Comment (baberlevi) [reply lib/spack/spack/test/package_sanity.py]: Sure, I'll make those changes. Thanks for the feedback

**Pull request merged**